examples/evaluators/golang/serving/main.go

```diff
@@ -0,0 +1,125 @@
+/*
+This is a sample Evaluator built using the Evaluator Harness. It evaluates
+multiple proposals and approves a subset of them. This sample demonstrates 
+how to build a basic Evaluator using the Evaluator Harness . This example 
+over-simplifies the actual evaluation decisions and hence should not be 
+used as is for a real scenario.
+
+Copyright 2018 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	harness "github.com/GoogleCloudPlatform/open-match/internal/harness/evaluator/golang"
+	"github.com/GoogleCloudPlatform/open-match/internal/pb"
+)
+
+func main() {
+	// This invoke the harness to set up the Evaluator. The harness abstracts
+	// fetching proposals ready for evaluation and the process of transforming
+	// approved proposals into results that Open Match can relay to the caller
+	// requesting for matches.
+	harness.RunEvaluator(Evaluate)
+}
+
+// Evaluate is where your custom evaluation logic lives.
+// Input:
+//  - proposals : List of all the proposals to be consiered for evaluation. Each proposal will have 
+//                Rosters comprising of the players belonging to that proposal. 
+// Output:
+//  - (proposals) : List of approved proposal IDs that can be returned as match results.
+func Evaluate(ctx context.Context, proposals []*pb.MatchObject) ([]string, error) {
+	// Map of approved and overloaded proposals. Using maps for easier lookup.
+	approvedProposals := make(map[string]bool)
+	overloadedProposals := make(map[string]bool)
+
+	// Map of all the players encountered in the proposals. Each entry maps a player id to
+	// the first match in which the player was encountered.
+	allPlayers := make(map[string]string)
+
+	// Iterate over each proposal to either add to approved map or overloaded map.
+	for _, proposal := range proposals {
+		proposalID := proposal.Id
+		approved := true
+		players := getPlayersInProposal(proposal)
+		// Iterate over each player in the proposal to check if the player was encountered before.
+		for _, pID := range players {
```

Comment:
Because there's player and proposal pID is ambiguous. playerID is preferred here.

Reply:
Good idea. Done!